### PR TITLE
disable the storage wrapper if we know there are no rules configured

### PR DESCRIPTION
**Pull request by icewind1991**

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -73,5 +78,21 @@ public function register(IRegistrationContext $context): void {
...
 
 	public function boot(IBootContext $context): void {
+		$this->hasAccessRules = $context->injectFn([$this, 'checkHasAccessRules']);
+	}
+
+	public function checkHasAccessRules(IManager $manager, IUserSession $userSession) {
+		$scopes[] = new ScopeContext(IManager::SCOPE_ADMIN);
+		$user = $userSession->getUser();
+		if ($user !== null && $manager->isUserScopeEnabled()) {
+			$scopes[] = new ScopeContext(IManager::SCOPE_USER, $user->getUID());
```
Comment (nickvergessen) [inline lib/AppInfo/Application.php:88]: At the moment this is not needed:
https://github.com/nextcloud/files_accesscontrol/blob/master/lib/Operation.php#L251-L253

So we could save a query on each request by not asking for it.

File: lib/AppInfo/Application.php [hunk 2]
```diff
@@ -73,5 +78,21 @@ public function register(IRegistrationContext $context): void {
 	}
 
 	public function boot(IBootContext $context): void {
+		$this->hasAccessRules = $context->injectFn([$this, 'checkHasAccessRules']);
+	}
+
+	public function checkHasAccessRules(IManager $manager, IUserSession $userSession) {
```
Comment (nickvergessen) [inline lib/AppInfo/Application.php:84]: Speaking of query per request, could we cache the boolean of this function in redis, etc?

Comment (icewind1991) [reply lib/AppInfo/Application.php]: Since the cache would need to be invalidated when rules are changed, this would need to be done in the workflowengine

**Review by nickvergessen — APPROVED** [review]
As mentioned in the chat. We can add this sure. But in such a case the app should just be disabled by the admin instead, to avoid more loading and a DB query (in case no other workflow app exists)

File: lib/AppInfo/Application.php [hunk 3]
```diff
@@ -26,15 +26,20 @@
 use OCA\FilesAccessControl\Listener\FlowRegisterOperationListener;
 use OCA\FilesAccessControl\Operation;
 use OCA\FilesAccessControl\StorageWrapper;
+use OCA\WorkflowEngine\Helper\ScopeContext;
 use OCP\AppFramework\App;
 use OCP\AppFramework\Bootstrap\IBootContext;
 use OCP\AppFramework\Bootstrap\IBootstrap;
 use OCP\AppFramework\Bootstrap\IRegistrationContext;
 use OCP\Files\Storage\IStorage;
+use OCP\IUserSession;
```
Comment (nickvergessen) [inline lib/AppInfo/Application.php:35]: Lint is unhappy

**Comment by nickvergessen** [comment]
Closing as per 
> As mentioned in the chat. We can add this sure. But in such a case the app should just be disabled by the admin instead, to avoid more loading and a DB query (in case no other workflow app exists)

**Pull request closed without merging**